src/shapley_sensitivity.jl

```diff
@@ -0,0 +1,327 @@
+@doc raw"""
+
+    Shapley(dim::Int; nboot::Int=500, n_perms::Int=-1,n_var::Int=1000, n_outer::Int=100, n_inner::Int=3) 
+
+- `dim`: number of features 
+- `nboot`: for confidence interval calculation `nboot` should be specified for the number (>0) of bootstrap runs.
+- `n_perms`: The number of permutations. If None, the exact permutations method is considerd.
+- `n_var`: The sample size for the output variance estimation.
+- `n_outer`: The number of conditionnal variance estimations.
+- `n_inner`: The sample size for the conditionnal output variance estimation.
+
+
+## Method Details
+
+Shapley values come from cooperative game theory and were introduced 1953. 
+They help determine the contribution of each player in the total payoff achieved by the coalition of the players. 
+The sum of Shapley effects over all individual variable is equal to the variance; this normalization property gives us a better interpretability in 
+determining the relative importance of variables [1, 2]. This also makes Shapley values more fair as they do not assign large importances to few inputs.
+Sobol indices assume that all the inputs are independent, which may be erroneous in many real-world scenarios like 
+physiology based pharmacokinetic models of the organs of the human body. 
+Shapley indices consider that parameters of a model can be correlated. 
+
+[1]:Owen, A. B., & Prieur, C. (2017). On Shapley value for measuring importance of dependent inputs. SIAM/ASA Journal on Uncertainty Quantification, 5(1), 986-1002.
+[2]:Song, E., Nelson, B. L., & Staum, J. (2016). Shapley effects for global sensitivity analysis: Theory and computation. SIAM/ASA Journal on Uncertainty Quantification, 4(1), 1060-1083.
+
+
+## API
+
+    gsa(f, method::Shapley, input_distribution::SklarDist; kwargs...)
+
+`input_distribution` is the joint distribution of the feature vectors.
+To create this, first define the marginal distirbutions (uniform distirbutions). Then define their 
+correlation using a copula (Copulas.jl can be used), Combine marginals and copula into a joint distirbution.
+### Example
+
+```julia
+using GlobalSensitivity, Copulas
+
+function ishi(X)
+    A= 7
+    B= 0.1
+    sin(X[1]) + A*sin(X[2])^2+ B*X[3]^4 *sin(X[1])
+end
+
+function ishi_batch(X)
+    A= 7
+    B= 0.1
+    @. sin(X[:,1]) + A*sin(X[:,2])^2+ B*X[:,3]^4 *sin(X[:,1])
+end
+
+margins = (Uniform(-pi, pi), Uniform(-pi, pi), Uniform(-pi, pi), Uniform(-pi, pi));
+dependency_matrix = [1 0 0 0; 0 1 0 0; 0 0 1 0; 0 0 0 1];
+dim = 4;
+C = GaussianCopula(dependency_matrix);
+input_distribution = SklarDist(C,margins);
+method = Shapley(dim);
+
+res1 = gsa(ishi,method,input_distribution,batch=false)
+
+res2 = gsa(ishi,method,input_distribution,batch=true)
+```
+"""
+
+## Data structures
+struct Shapley <: GSAMethod
+    nboot::Int
+    n_perms::Int
```

Comment:
Typically you'd want to set it to `nothing` as the default instead of -1, though it doesn't matter much here for now